File: javascript/sorting/bogoSort.js
function IsSorted(list){
    n = list.length
    for(i=0; i<n-1; i++){
        if(list[i] > list[i + 1])
        {
            return false
        }
    } return true
}

function bogoSort(arr) {
    var count = 0
    while(!IsSorted(arr)){
        arr.sort(() => Math.random() - 0.5);
        count += 1
    }
    return [arr, count]
}

let arr = [13, 52, 33, 27, 68];
var b = bogoSort(arr);
console.log(b)

File: python/sorting/bogo_sort.py
import random

def is_sorted(list):
    for index in range(len(list)-1):
        if list[index] > list[index + 1]:
            return False
    return True

def bogo_sort(list):
    count = 0
    while not is_sorted(list):
        random.shuffle(list)
        count += 1
    return list,count

alist = [1,4,10,6,7]
alist2 = [3,4,5,6,7]
print(bogo_sort(alist))
print(bogo_sort(alist2))